tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.h

```diff
@@ -62,17 +62,17 @@ KDL::Frame gmTransformToKDL(const geometry_msgs::TransformStamped& t)
 // method to extract timestamp from object
 template <>
 inline
-  const builtin_interfaces::msg::Time& getTimestamp(const geometry_msgs::Vector3Stamped& t) {return t.header.stamp;}
+  const tf2::TimePoint& getTimestamp(const geometry_msgs::msg::Vector3Stamped& t) {return tf2_ros::fromMsg(t.header.stamp);}
```

Comment:
The return of a reference in this case doesn't work since the local variable is being destroyed. when the function scope is left. So this code results in accessing invalid memory.

Same below for all occasions with the same signature.